src/devices/PiJuice/PiJuiceStatus.cs

```diff
@@ -0,0 +1,275 @@
+
+            var buttonEvents = new List<ButtonEventType>(3)
+            {
+                GetButtonEventType(response[0] & 0x0F),
+                GetButtonEventType((response[0] >> 4) & 0x0F),
+                GetButtonEventType(response[1] & 0x0F)
+            };
+
+            return buttonEvents;
+        }
+
+        /// <summary>
+        /// Clears generated button event
+        /// </summary>
+        /// <param name="button">Button to clear button event for</param>
+        public void ClearButtonEvent(Button button)
+        {
+            var array = button switch
+            {
+                Button.SW1 => new byte[] { 0xF0, 0xFF },
+                Button.SW2 => new byte[] { 0x0F, 0xFF },
+                Button.SW3 => new byte[] { 0xFF, 0xF0 },
+                _ => throw new NotImplementedException()
+            };
+
+            _piJuice.WriteCommand(PiJuiceCommand.ButtonEvent, array);
+        }
+
+        /// <summary>
+        /// Get battery charge level between 0 and 100 percent
+        /// </summary>
+        /// <returns>Battery charge level percentage</returns>
+        public byte GetChargeLevel()
+        {
+            var response = _piJuice.ReadCommand(PiJuiceCommand.ChargeLevel, 1);
+
+            return response[0];
+        }
+
+        /// <summary>
+        /// Get battery temperature
+        /// </summary>
+        /// <returns>Battery temperature in celsius</returns>
+        public Temperature GetBatteryTemperature()
+        {
+            var response = _piJuice.ReadCommand(PiJuiceCommand.BatteryTemperature, 2);
+
+            return new Temperature((response[1] << 8) | response[0], TemperatureUnit.DegreeCelsius);
+        }
+
+        /// <summary>
+        /// Get battery voltage
+        /// </summary>
+        /// <returns>Battery voltage in millivolts</returns>
+        public ElectricPotential GetBatteryVoltage()
+        {
+            var response = _piJuice.ReadCommand(PiJuiceCommand.BatteryVoltage, 2);
+
+            return new ElectricPotential((response[1] << 8) | response[0], ElectricPotentialUnit.Millivolt);
+        }
+
+        /// <summary>
+        /// Get battery current
+        /// </summary>
+        /// <returns>Battery current in milliamps</returns>
+        public ElectricCurrent GetBatteryCurrent()
+        {
+            var response = _piJuice.ReadCommand(PiJuiceCommand.BatteryCurrent, 2);
+
+            var i = (response[1] << 8) | response[0];
+            if ((i & (1 << 15)) == (1 << 15))
+            {
+                i -= (1 << 16);
+            }
+
+            return new ElectricCurrent(i, ElectricCurrentUnit.Milliampere);
+        }
+
+        /// <summary>
+        /// Get supplied voltage
+        /// </summary>
+        /// <returns>Voltage supplied from the GPIO power output from the PiJuice or when charging, voltage supplied in millivolts</returns>
+        public ElectricPotential GetIOVoltage()
```

Comment:
`GetIoVoltage` most likely. Ssomehow for tIO I'll be ok with either IO either Io. But it has to be consistent everyone. Still a preference for Io.

Reply:
I'm going to stick with IO on the bases that it is used in the exception IOException which comes from the GrovePi, which is what I based this project on.

Reply:
Fair enough. Go then for IO. Please check everywhere to be consistent then.